### ISSUE TITLE
Autoencoder tutorial not reproducible

### ISSUE THREAD

**Issue by jonathanronen** [fchollet/keras#3642]
The tutorial on the keras blog has some easy steps to build an autoencoder on MNIST
https://blog.keras.io/building-autoencoders-in-keras.html

When I try to reproduce the first step, I get quite different (like way worse) reconstructions:
https://gist.github.com/jonathanronen/9b339c4a1816b61b7ffeeab88df3a6d6

Code is copied and pasted from the tutorial.

**Comment by fchollet** [comment]
It may seem obvious, but you should not be feeding integers in the 0-255 range in a NN. You need to cast your data to float and normalize it.

**Comment by jonathanronen** [comment]
I see. That's the part I left off. Thanks @fchollet!
